microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+
+Traces recorded by the Edge tracing tool provide a lot of information about the browser. To filter the information to only what's relevant to the webpage that your extension impacted, use the following steps:
+
+1. Press **Shift+Esc** to open the **Browser Task Manager** dialog box.
+
+1. In the **Browser Task Manager** dialog box, search for the tab that corresponds to the webpage that your extension affected, and note the number in the **Process ID** column. Close the dialog box.
+
+1. In the toolbar of the Edge tracing tool, click **Processes**, and then select the checkbox that corresponds to the process ID you noted. Clear all other checkboxes.
+
+1. In the top right corner of the Edge tracing tool, click the search field, type **ScriptInjection::InjectJS**, and then press **Enter** until an event that corresponds to your extension gets highlighted in the bottom panel.
+
+   The bottom panel displays the start time and total duration of the event.
+
+   ![The Edge tracing tool showing a scrip injection event](./minimize-page-load-time-impact-images/tracing-event-selected.png)
+
+To continue analyzing the performance impact of your extension's content script on the webpage, look for the following key events inside the **ScriptInjection::InjectJS** event:
+
+* **v8.compile**: this event shows the compilation time of your content script.
+* **v8.run**: this event indicates the running time of the compiled script.
+
+
+<!-- ====================================================================== -->
+## Only add content script code that's needed for the functionality of your extension
+
+Your extension's content script runs in the context of the webpage. To minimize the impact of your content script on that webpage, make sure to only add, in your content script, the minimum amount of code that your extension needs to run in the context of the webpage. Audit the code in your content script and remove legacy frameworks, tools, libraries, or other code that aren't needed by your content script to run in Microsoft Edge.
+
+You can make use of lazy loading and code splitting techniques to minimize the amount of code that runs in your content script:
+
+* Lazy loading is the process of loading is the process of loading your code only when it's needed, based on the user actions, the page content, or the extension logic.
+* Code splitting is the process of dividing your code into smaller chunks, or modules, which can be loaded separately or on demand.
+
+If your extension is small enough, you won't need a build tool to split your code. If your extension is larger, and if your code is more complex to manage, use a build to split your code into smaller chunks. Build tools can help you organize your code into logical units, which can be loaded on demand. For example, you can use webpack to split your code into _entry points_ and _dynamic imports_:
```

Comment:
```suggestion
If your extension is small enough, you won't need a build tool to split your code. If your extension is larger, and if your code is more complex to manage, use a build tool to split your code into smaller chunks. Build tools can help you organize your code into logical units, which can be loaded on demand. For example, you can use webpack to split your code into _entry points_ and _dynamic imports_:
```
"use a build" - ??  means use a build _tool_?  @captainbrosset

Reply:
Yes, this should be "build tool"